ShopCart/Program1.js

```diff
@@ -5,67 +5,95 @@ app.controller('TestCtrl', [ '$scope',
     }
 ]);
 
-app.directive('testDirec', function () {
-    
-    
-    function linkingFunction(scope, element, attrs) {
-    }
+app.directive('product', function () {
+ 
+    function linkingFunction(scope, element, attrs) {}
 
     function controller($scope) {
-        
-        $scope.productList=[
+              $scope.productList=[
         {
             id:0,
             name:"product0",
             price:34,
-            qty:4,
+            qty:1,
         },
         {
             id:1,
             name:"product1",
             price:93,
-            qty:4,
+            qty:1,
         },
         {
             id:3,
             name:"product3",
             price:4,
-            qty:23,
+            qty:1,
         }
-        ];
-        ///list of items on the cart        
-        $scope.cartList=[
-            {
-            id:1,
-            name:"product1",
-            price:93,
-            qty:2,
-            }];
-                
-        //adds to shopping cart
-        $scope.addToCart=function(product){
-            var idSearch="qty"+product.id;
-            var qty=Number(document.getElementById(idSearch).value);
+        ];   
+      $scope.getQty=function(product){
+           var idSearch="qty"+product.id;
+           $scope.qty= Number(document.getElementById(idSearch).value);   
+        }   
+        
+    }
+    return {
+        restrict: 'E'
+        , template:  '<div name="ProductForm">'+
+        '<div ng-model"listOflements">'+
+        '<h3 class="titleMain">Product List</h3>'+
+                '<div class="product" ng-repeat="product in productList">'+
+                    '<b class="title">{{product.name}} </b><br>'+
+                    '<b>Price:$</b>{{product.price}}'+
+        '<div>qty:<input type="text" " style="width:20px;" value="1" id="qty{{product.id}}" ng-model"qty" ></div>'+
+        '<div><input type="submit" value="Add to Cart"  ng-click="getQty(product);addToCart(product,qty)"> </div>'+
+                '</div>'+
+       ' </div>'
+        , scope: {
+            addToCart:'='
+        }
+        , link: linkingFunction
+        , controller: controller
+    }
+});
+
+app.directive('cart', function () {
+ 
+    function linkingFunction(scope, element, attrs) {
+        scope.addToCart=function(product,qty){
```

Comment:
This is entirely logic based and does not belong in the linking function. Only DOM manipulation should occur in the linking function.  Move this stuff to the controller.